api/cockpit/v1beta1/cockpit_utils.go

```diff
@@ -0,0 +1,62 @@
+package cockpit
+
+import (
+	"time"
+
+	"github.com/scaleway/scaleway-sdk-go/internal/async"
+	"github.com/scaleway/scaleway-sdk-go/internal/errors"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	defaultRetryInterval = 3 * time.Second
+	defaultTimeout       = 5 * time.Minute
+)
+
+// WaitForCockpitRequest is used by WaitForCockpit method.
+type WaitForCockpitRequest struct {
+	ProjectID     string
+	Timeout       *time.Duration
+	RetryInterval *time.Duration
+}
+
+func (s *API) WaitForCockpit(
+	req *WaitForCockpitRequest,
+	opts ...scw.RequestOption,
+) (*Cockpit, error) {
+	timeout := defaultTimeout
+	if req.Timeout != nil {
+		timeout = *req.Timeout
+	}
+	retryInterval := defaultRetryInterval
+	if req.RetryInterval != nil {
+		retryInterval = *req.RetryInterval
+	}
+
+	terminalStatus := map[CockpitStatus]struct{}{
+		CockpitStatusReady: {},
+		CockpitStatusError: {},
+	}
```

Comment:
I suggest we eventually harmonize with the sdk-js/python logic, which is "wait until we see transient statutes". I think it would lead to less infinite loop in case the code or the API response is not up to date.

But probably for when we'll generate these helpers :) 

Besides that, lgtm.